### ISSUE TITLE
no app could be started / no file browse possible after restore 

### ISSUE THREAD

**Issue by michaelof** [nextcloud/server#20863]
My Nextcloud was affected by this issue: https://help.nextcloud.com/t/nextcloud-16-password-change-is-disabled-because-the-master-key-is-disabled/56022/8
So I tried the described workaround up to the step "occ encryption:enable-master-key"
Result: All files are "available", but in fact encrypted data garbage.

So better to take occ warning serious, and I've had to restore my backup from last day.

Peformed the restore exactly as described in Admin Guide, no errors.

Result: Login is possible, but Browser (more or less) hangs forever. No files are showm, no contacts, no ..., everything very slow. Tried with another browser to exclude a browser issue.

Issue already described and disccussed here: https://help.nextcloud.com/t/nexcloud-not-avialable-after-restore/80764

### Server configuration
OpenSuse 15.1
Apache 2.4.33
Mariadb 10.2.31
PHP 7.2.5
Nextcloud 18.3
Update for a very long time, originally from owncloud
Install from OpenSuse repos

**List of activated apps:**
Enabled:
  - accessibility: 1.4.0
  - activity: 2.11.0
  - admin_audit: 1.8.0
  - audioplayer: 2.10.0
  - bruteforcesettings: 1.6.0
  - calendar: 2.0.3
  - cloud_federation_api: 1.1.0
  - comments: 1.8.0
  - contacts: 3.3.0
  - dav: 1.14.0
  - encryption: 2.6.0
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_external: 1.9.0
  - files_pdfviewer: 1.7.0
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - files_videoplayer: 1.7.0
  - firstrunwizard: 2.7.0
  - logreader: 2.3.0
  - lookup_server_connector: 1.6.0
  - music: 0.14.0
  - nextcloud_announcements: 1.7.0
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - password_policy: 1.8.0
  - photos: 1.0.0
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - recommendations: 0.6.0
  - serverinfo: 1.8.0
  - settings: 1.0.0
  - sharebymail: 1.8.0
  - spreed: 8.0.8
  - support: 1.1.0
  - survey_client: 1.6.0
  - systemtags: 1.8.0
  - tasks: 0.12.1
  - text: 2.0.0
  - theming: 1.9.0
  - twofactor_backupcodes: 1.7.0
  - twofactor_totp: 4.1.3
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
Disabled:
  - mail
  - richdocuments
  - user_ldap

**Nextcloud configuration:**
 sudo -u wwwrun php occ config:list
The current PHP memory limit is below the recommended value of 512MB.
Nextcloud is in maintenance mode - no apps have been loaded

{
    "system": {
        "updatechecker": false,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost:10080",
            "cloud.example.com",
            "cloudv4.example.com",
            "www.lateinklasse.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/localhost:10080\/nextcloud",
        "dbtype": "mysql",
        "version": "18.0.4.2",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_smtpmode": "sendmail",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "maintenance": true,
        "appstore.experimental.enabled": true,
        "theme": "",
        "loglevel": 2,
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "log_rotate_size": 10485760,
        "app_install_overwrite": [
            "tasks",
            "contacts",
            "unsplash",
            "bruteforcesettings"
        ],
        "mysql.utf8mb4": true,
        "data-fingerprint": "00699d79ff00538dcec499b4e6d1e974"
    },
    "apps": {
        "accessibility": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "types": ""
        },
        "activity": {
            "enabled": "yes",
            "installed_version": "2.11.0",
            "signed": "true",
            "types": "filesystem"
        },
        "admin_audit": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "logging"
        },
        "audioplayer": {
            "enabled": "yes",
            "installed_version": "2.10.0",
            "types": "filesystem"
        },
        "backgroundjob": {
            "lastjob": "2242"
        },
        "bruteforcesettings": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": ""
        },
        "calendar": {
            "enabled": "yes",
            "installed_version": "2.0.3",
            "signed": "true",
            "types": ""
        },
        "cloud_federation_api": {
            "enabled": "yes",
            "installed_version": "1.1.0",
            "types": "filesystem"
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "logging"
        },
        "configreport": {
            "enabled": "no",
            "installed_version": "0.1.1",
            "types": "filesystem"
        },
        "contacts": {
            "enabled": "yes",
            "installed_version": "3.3.0",
            "signed": "true",
            "types": ""
        },
        "core": {
            "backgroundjobs_mode": "cron",
            "default_encryption_module": "OC_DEFAULT_MODULE",
            "encryption_enabled": "yes",
            "enterpriseLogoChecked": "yes",
            "incoming_server2server_share_enabled": "no",
            "installed.bundles": "[\"CoreBundle\"]",
            "installedat": "1471639347.995",
            "lastcron": "1588854602",
            "lastupdateResult": "[]",
            "lastupdatedat": "1588822265",
            "moveavatarsdone": "yes",
            "oc.integritycheck.checker": "[]",
            "outgoing_server2server_share_enabled": "no",
            "previewsCleanedUp": "1",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "repairlegacystoragesdone": "yes",
            "scss.variables": "8b2da810f59e0aaaa00f167967072ac1",
            "shareapi_allow_group_sharing": "yes",
            "shareapi_allow_mail_notification": "yes",
            "shareapi_allow_public_notification": "yes",
            "shareapi_allow_share_dialog_user_enumeration": "yes",
            "shareapi_default_expire_date": "yes",
            "shareapi_enforce_expire_date": "yes",
            "shareapi_enforce_links_password": "no",
            "shareapi_exclude_groups": "no",
            "shareapi_exclude_groups_list": "[\"\"]",
            "shareapi_expire_after_n_days": "30",
            "shareapi_only_share_with_group_members": "yes",
            "theming.variables": "d41d8cd98f00b204e9800998ecf8427e",
            "umgmt_send_email": "true",
            "umgmt_show_backend": "true",
            "umgmt_show_email": "true",
            "umgmt_show_last_login": "true",
            "umgmt_show_storage_location": "true",
            "vendor": "nextcloud"
        },
        "dav": {
            "buildCalendarReminderIndex": "yes",
            "buildCalendarSearchIndex": "yes",
            "chunks_migrated": "1",
            "enabled": "yes",
            "installed_version": "1.14.0",
            "regeneratedBirthdayCalendarsForYearFix": "yes",
            "sendEventRemindersPush": "no",
            "types": "filesystem"
        },
        "drop_account": {
            "enabled": "no",
            "installed_version": "0.0.6",
            "types": ""
        },
        "encryption": {
            "enabled": "yes",
            "installed_version": "2.6.0",
            "masterKeyId": "master_c768c091",
            "publicShareKeyId": "pubShare_c768c091",
            "recoveryAdminEnabled": "1",
            "recoveryKeyId": "recoveryKey_c768c091",
            "types": "filesystem",
            "useMasterKey": "0"
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": ""
        },
        "federation": {
            "autoAddServers": "0",
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "default_quota": "20 GB",
            "enabled": "yes",
            "installed_version": "1.13.1",
            "types": "filesystem"
        },
        "files_external": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "signed": "true",
            "types": ""
        },
        "files_rightclick": {
            "enabled": "yes",
            "installed_version": "0.15.2",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "incoming_server2server_group_share_enabled": "yes",
            "incoming_server2server_share_enabled": "yes",
            "installed_version": "1.10.1",
            "lookupServerUploadEnabled": "no",
            "outgoing_server2server_group_share_enabled": "yes",
            "outgoing_server2server_share_enabled": "yes",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "no",
            "installed_version": "2.8.0",
            "signed": "true",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "filesystem,dav"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.11.0",
            "types": "filesystem,dav"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "yes",
            "installed_version": "2.7.0",
            "types": "logging"
        },
        "gallery": {
            "enabled": "no",
            "installed_version": "18.4.0",
            "signed": "true",
            "types": ""
        },
        "issuetemplate": {
            "enabled": "no",
            "installed_version": "0.6.0",
            "types": ""
        },
        "jsloader": {
            "cachebuster": "3",
            "enabled": "no",
            "installed_version": "1.0.2",
            "snippet": "",
            "types": "",
            "url": ""
        },
        "logreader": {
            "enabled": "yes",
            "installed_version": "2.3.0",
            "types": ""
        },
        "lookup_server_connector": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "authentication"
        },
        "mail": {
            "enabled": "no",
            "installed_version": "1.3.3",
            "types": ""
        },
        "market": {
            "enabled": "no",
            "installed_version": "0.2.1",
            "signed": "true",
            "types": ""
        },
        "music": {
            "enabled": "yes",
            "installed_version": "0.14.0",
            "ocsid": "164319",
            "types": "filesystem"
        },
        "nextcloud_announcements": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "pub_date": "Thu, 24 Oct 2019 00:00:00 +0200",
            "types": "logging"
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "2.6.0",
            "types": "logging"
        },
        "oauth2": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": "authentication"
        },
        "password_policy": {
            "enabled": "yes",
            "enforceHaveIBeenPwned": "0",
            "enforceNumericCharacters": "0",
            "enforceSpecialCharacters": "0",
            "enforceUpperLowerCase": "0",
            "installed_version": "1.8.0",
            "types": ""
        },
        "photos": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": ""
        },
        "polls": {
            "enabled": "no",
            "installed_version": "0.10.4",
            "types": ""
        },
        "privacy": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "readableLocation": "de",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "prevent_group_restriction"
        },
        "recommendations": {
            "enabled": "yes",
            "installed_version": "0.6.0",
            "types": ""
        },
        "richdocuments": {
            "enabled": "no",
            "installed_version": "3.5.3",
            "signed": "true",
            "types": "filesystem,dav,prevent_group_restriction"
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": ""
        },
        "settings": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": ""
        },
        "sharebymail": {
            "enabled": "yes",
            "enforcePasswordProtection": "yes",
            "installed_version": "1.8.0",
            "types": "filesystem"
        },
        "spreed": {
            "enabled": "yes",
            "installed_version": "8.0.8",
            "project_access_invalidated": "1",
            "signaling_ticket_secret": "***REMOVED SENSITIVE VALUE***",
            "stun_servers": "***REMOVED SENSITIVE VALUE***",
            "types": "prevent_group_restriction"
        },
        "spreedme": {
            "SPREED_WEBRTC_SHAREDSECRET": "33a0d7f6adf6af5793504926d710752373e35b74dadc2fdad0114e55a6f0e980",
            "enabled": "no",
            "installed_version": "0.3.11",
            "types": ""
        },
        "support": {
            "enabled": "yes",
            "installed_version": "1.1.0",
            "types": "session"
        },
        "survey_client": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "last_report": "{\"id\":\"ocv90mja2d89\",\"items\":[[\"server\",\"version\",\"12.0.5.3\"],[\"server\",\"code\",\"other\"],[\"server\",\"enable_avatars\",\"yes\"],[\"server\",\"enable_previews\",\"yes\"],[\"server\",\"memcache.local\",\"none\"],[\"server\",\"memcache.distributed\",\"none\"],[\"server\",\"asset-pipeline.enabled\",\"no\"],[\"server\",\"filelocking.enabled\",\"yes\"],[\"server\",\"memcache.locking\",\"none\"],[\"server\",\"debug\",\"no\"],[\"server\",\"cron\",\"cron\"],[\"php\",\"version\",\"7.0.7\"],[\"php\",\"memory_limit\",536870912],[\"php\",\"max_execution_time\",3600],[\"php\",\"upload_max_filesize\",535822336],[\"database\",\"type\",\"mysql\"],[\"database\",\"version\",\"10.0.32\"],[\"database\",\"size\",12140544],[\"apps\",\"activity\",\"2.5.2\"],[\"apps\",\"audioplayer\",\"2.2.4\"],[\"apps\",\"bruteforcesettings\",\"1.0.3\"],[\"apps\",\"calendar\",\"1.5.7\"],[\"apps\",\"comments\",\"1.2.0\"],[\"apps\",\"configreport\",\"0.1.1\"],[\"apps\",\"contacts\",\"2.0.1\"],[\"apps\",\"dav\",\"1.3.1\"],[\"apps\",\"drop_account\",\"disabled\"],[\"apps\",\"encryption\",\"1.6.0\"],[\"apps\",\"federatedfilesharing\",\"1.2.0\"],[\"apps\",\"federation\",\"1.2.0\"],[\"apps\",\"files\",\"1.7.2\"],[\"apps\",\"files_external\",\"1.3.0\"],[\"apps\",\"files_pdfviewer\",\"1.1.1\"],[\"apps\",\"files_sharing\",\"1.4.0\"],[\"apps\",\"files_texteditor\",\"2.4.1\"],[\"apps\",\"files_trashbin\",\"1.2.0\"],[\"apps\",\"files_versions\",\"1.5.0\"],[\"apps\",\"files_videoplayer\",\"1.1.0\"],[\"apps\",\"firstrunwizard\",\"2.1\"],[\"apps\",\"gallery\",\"17.0.0\"],[\"apps\",\"logreader\",\"2.0.0\"],[\"apps\",\"lookup_server_connector\",\"1.0.0\"],[\"apps\",\"market\",\"0.2.1\"],[\"apps\",\"music\",\"0.5.5\"],[\"apps\",\"nextcloud_announcements\",\"1.1\"],[\"apps\",\"notifications\",\"2.0.0\"],[\"apps\",\"oauth2\",\"1.0.5\"],[\"apps\",\"password_policy\",\"1.2.2\"],[\"apps\",\"provisioning_api\",\"1.2.0\"],[\"apps\",\"richdocuments\",\"disabled\"],[\"apps\",\"serverinfo\",\"1.2.0\"],[\"apps\",\"sharebymail\",\"1.2.0\"],[\"apps\",\"spreed\",\"2.0.2\"],[\"apps\",\"spreedme\",\"disabled\"],[\"apps\",\"survey_client\",\"1.0.0\"],[\"apps\",\"systemtags\",\"1.2.0\"],[\"apps\",\"tasks\",\"0.9.5\"],[\"apps\",\"templateeditor\",\"0.1\"],[\"apps\",\"theming\",\"1.3.0\"],[\"apps\",\"twofactor_backupcodes\",\"1.1.1\"],[\"apps\",\"updatenotification\",\"1.2.0\"],[\"apps\",\"workflowengine\",\"1.2.0\"],[\"stats\",\"num_files\",3776],[\"stats\",\"num_users\",2],[\"stats\",\"num_storages\",4],[\"stats\",\"num_storages_local\",2],[\"stats\",\"num_storages_home\",2],[\"stats\",\"num_storages_other\",0],[\"stats\",\"num_comments\",0],[\"stats\",\"num_comment_markers\",0],[\"stats\",\"num_systemtags\",0],[\"stats\",\"num_systemtags_mappings\",0],[\"files_sharing\",\"num_shares\",2],[\"files_sharing\",\"num_shares_user\",0],[\"files_sharing\",\"num_shares_groups\",0],[\"files_sharing\",\"num_shares_link\",2],[\"files_sharing\",\"num_shares_link_no_password\",2],[\"files_sharing\",\"num_fed_shares_sent\",0],[\"files_sharing\",\"num_fed_shares_received\",0],[\"files_sharing\",\"permissions_3_1\",\"2\"],[\"encryption\",\"enabled\",\"yes\"]]}",
            "last_sent": "1516831315",
            "types": ""
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "logging"
        },
        "tasks": {
            "enabled": "yes",
            "installed_version": "0.12.1",
            "signed": "true",
            "types": ""
        },
        "templateeditor": {
            "enabled": "no",
            "installed_version": "0.1",
            "types": ""
        },
        "text": {
            "enabled": "yes",
            "installed_version": "2.0.0",
            "types": "dav"
        },
        "theming": {
            "cachebuster": "78",
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": "logging"
        },
        "twofactor_backupcodes": {
            "enabled": "yes",
            "installed_version": "1.7.0",
            "types": ""
        },
        "twofactor_totp": {
            "enabled": "yes",
            "installed_version": "4.1.3",
            "types": ""
        },
        "updatenotification": {
            "audioplayer": "2.10.0",
            "bruteforcesettings": "1.6.0",
            "calendar": "2.0.2",
            "contacts": "3.3.0",
            "core": "18.0.4.2",
            "enabled": "yes",
            "files_pdfviewer": "1.2.1",
            "files_rightclick": "0.15.1",
            "installed_version": "1.8.0",
            "issuetemplate": "0.4.0",
            "mail": "1.3.4",
            "music": "0.14.0",
            "richdocuments": "3.5.3",
            "spreed": "8.0.8",
            "tasks": "0.12.2",
            "theming": "1.4.5",
            "types": "",
            "update_check_errors": "0"
        },
        "viewer": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": ""
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "2.0.0",
            "types": "filesystem"
        }
    }
}

**Are you using external storage, if yes which one:** NO

**Are you using encryption:** YES

**Are you using an external user-backend, if yes which one:** Webdav

### Client configuration
**Browser:**
firefox, linux/win10/android, chromium, wget/curl, ...

**Operating system:**
mainly linux

#### Web server error log
nothing special in access_log, all nextcloud stuff usually http/200
nothing for days in error_log

#### Nextcloud log (data/nextcloud.log)
Link to nextcloud log filtered by "chrome" here: https://paste.opensuse.org/56462028

**Comment by kesselb** [comment]
`include(/srv/www/htdocs/nextcloud/apps/files_trashbin/composer/composer/../../lib/Trash/TrashManager.php): failed to open stream: No such file or directory`

Means: That (and probably more files) are missing.

**Comment by michaelof** [comment]
wow, you're completely right, missing!

Currently no "TrashManager.php" on my VPS, in no folder.

What I've checked and found out after your very valuable hint:

1. pulled nextcloud latest docker image, looked inside:

`# find / -name TrashManager.php -print 2>/dev/null
/usr/src/nextcloud/apps/files_trashbin/lib/Trash/TrashManager.php
/var/www/html/apps/files_trashbin/lib/Trash/TrashManager.php
`

2. checked content of OpenSuse's nextcloud rpm, which I've used for installation, works fine and lightspeed fast with OpenSuse rpms for years now:

`# rpm -ql nextcloud | grep Trashbin
/srv/www/htdocs/nextcloud/apps/admin_audit/lib/Actions/Trashbin.php
/srv/www/htdocs/nextcloud/apps/files_trashbin/lib/Trashbin.php
`

so SHOULD be there.

3. checked content on my local backup device:

`localmachine:/run/media/michael/USB_HDD/vServerSnapshots/daily.0/vserver.example.com/srv/www/htdocs/nextcloud/apps/files_trashbin/lib # ls -l
total 60
drwxrwxr-x 2 wwwrun www 4096 Apr 23 21:36 AppInfo
drwxrwxr-x 2 wwwrun www 4096 Apr 23 21:36 BackgroundJob
-rw-rw-r-- 6 wwwrun www 1200 Apr 22 21:23 Capabilities.php
drwxrwxr-x 2 wwwrun www 4096 Apr 23 21:36 Command
drwxrwxr-x 2 wwwrun www 4096 Apr 23 21:36 Controller
drwxrwxr-x 2 wwwrun www 4096 Apr 23 21:36 Events
drwxrwxr-x 2 wwwrun www 4096 Apr 23 21:36 Exceptions
-rw-rw-r-- 6 wwwrun www 4986 Apr 22 21:23 Expiration.php
-rw-rw-r-- 6 wwwrun www 4607 Apr 22 21:23 Helper.php
-rw-rw-r-- 6 wwwrun www 1503 Apr 22 21:23 Hooks.php
drwxrwxr-x 2 wwwrun www 4096 Apr 23 21:36 Sabre
-rw-rw-r-- 6 wwwrun www 6222 Apr 22 21:23 Storage.php
`

FYI I'm using (rsync-based) rsnapshot for years to backup my VPS remotely via my local machine, to an usb hdd. Mainly daily, not always. But always before critical updates etc. :-) 

rsnapshot stores backups in the "...../daily.#/......." folders (see above), and intensively uses hard links for all unchanged files, so I looked also not only below the (current) daily.0 folder, but also in the daily.10, ... and even in a more than one years old (named backup) "lastOpenSuseLeap15.0" folder: 
RESULT:  ALWAYS same content, NO Trashbin.php

4. checked if there might be anything "special" with Trashbin.php, but seems to be no symbolic link or similar

5. checked my /etc/rsnapshot.conf settings and logs

As you see, e.g. "Storage.php" exists in same folder "Trashbin.php" does not. And as I (of course) have not excluded any *.php files manually, I've checked for the last rsync which fetched "Storage.php" to my local machine, was on March 14th, I guess nextcloud 18.0.4 update, not sure.

rsnapshot logs the rsync command performed:

`/usr/bin/rsync -azv --delete --numeric-ids --relative --delete-excluded \
    --exclude-from=/home/michael/BACKUP/exclude_list --rsh=/usr/bin/ssh \
    --link-dest=/run/media/michael/USB_HDD/vServerSnapshots/daily.0/vserver.example.com/ \
    root@vserver.example.com:/lib/ \
    /run/media/michael/USB_HDD/vServerSnapshots/.sync/vserver.example.com/
`

So NO IDEA why "Trashbin.php" has NEVER been included in backup.
Any hints would be great....

**Comment by michaelof** [comment]
Thinking and thinking about what might be "mystic" with this "Trashbin.php" file I've made one last test:

1. force reinstall of nextcloud rpm: "Trashbin.app" is now existing, at the expected locations
2. switched off nextcloud maintenance mode: Nextcloud works at once, biut with an uncertain state because my additional issues with encryption. But light at the end of the tunnel...
3. performed a rsnapshot backup
4. checked backup location: "Trashbin.app" ** NOT ** backed up
5. Verified that still there in (unchanged) nextcloud subfolders

Having found out this, not a nextcloud issue, I've looked again at my rsnapshot config. 

And, viola, found an entry in exclude file 
- [Tt]rash
probably a default, missed by me, long ago.....

- deleted this exclude line, rsnapshot-ed again, and (of course ;-)) Trashbin.php, along with several other [Tt]rash file and folders, is now included. Now and from now on :-S

Currently, to be sure, I'm again making a "fresh" restore, db and files, and after that I'll perform at once a "force reinstall rpm". Should be fine then.

@kesselb Thank you / Ganz herzlichen Dank !!!
